lib/ddtrace/profiling/scheduler.rb

```diff
@@ -0,0 +1,66 @@
+require 'ddtrace/worker'
+require 'ddtrace/workers/polling'
+
+module Datadog
+  module Profiling
+    # Pulls and exports profiling data on an async interval basis
+    class Scheduler < Worker
+      include Workers::Polling
+
+      attr_reader \
+        :exporters,
+        :recorder
+
+      def initialize(recorder, exporters, options = {})
+        @recorder = recorder
+        @exporters = [exporters].flatten
+
+        # Workers::Async::Thread settings
+        # Restart in forks by default
+        self.fork_policy = options.fetch(:fork_policy, Workers::Async::Thread::FORK_POLICY_RESTART)
+
+        # Workers::IntervalLoop settings
+        self.loop_base_interval = options[:interval] if options.key?(:interval)
+        self.loop_back_off_ratio = options[:back_off_ratio] if options.key?(:back_off_ratio)
+        self.loop_back_off_max = options[:back_off_max] if options.key?(:back_off_max)
+
+        # Workers::Polling settings
+        self.enabled = options.fetch(:enabled, false)
+      end
+
+      def start
+        perform
+      end
+
+      def perform
+        flush_events
+      end
+
+      def after_fork
+        # Clear recorder's buffers by flushing events.
+        # Objects from parent process will copy-on-write,
+        # and we don't want to send events for the wrong process.
+        recorder.pop
```

Comment:
Per the comment, its important to clear the buffer after forking occurs. This function will be invoked when `Async::Thread` restarts the worker thread after a fork.